routes/post-list/stage.tsx

```diff
@@ -0,0 +1,372 @@
+			} ),
+		];
+	}, [ postTypeActions ] );
+
+	const handleTabChange = useCallback(
+		( status: string ) => {
+			navigate( {
+				to: `/types/${ postType }/list/${ status }`,
+			} );
+		},
+		[ navigate, postType ]
+	);
+
+	if ( ! postTypeObject ) {
+		return null;
+	}
+
+	const selection = searchParams.postIds ?? [];
+
+	// Auto-select first post in list view if none selected
+	if ( view.type === 'list' && selection.length === 0 && posts?.length > 0 ) {
+		selection.push( posts[ 0 ].id.toString() );
+	}
+
+	// Until list view supports multi selection, only keep the first item.
+	if ( view.type === 'list' ) {
+		selection.splice( 1 );
+	}
+
+	return (
+		<Page
+			title={ postTypeObject.labels?.name }
+			subTitle={ postTypeObject.labels?.description }
+			className={ `${ postTypeObject.name.toLowerCase() }-page` }
+			actions={
+				<>
+					{ isModified && (
+						<Button
+							variant="tertiary"
+							size="compact"
+							onClick={ onReset }
+						>
+							{ __( 'Reset view' ) }
+						</Button>
+					) }
+					{ labels?.add_new_item &&
+						canCreateRecord &&
+						postType !== 'attachment' && (
+							<Button
+								variant="primary"
+								onClick={ () => {
+									navigate( {
+										to: `/types/${ postType }/new`,
+									} );
+								} }
+								size="compact"
+							>
+								{ labels.add_new_item }
+							</Button>
+						) }
+				</>
+			}
+			hasPadding={ false }
+		>
+			{ DEFAULT_VIEWS.length > 1 && (
+				<div className="routes-post-list__tabs-wrapper">
+					<Tabs
+						onSelect={ handleTabChange }
+						selectedTabId={ slug ?? 'all' }
+					>
+						<Tabs.TabList>
+							{ DEFAULT_VIEWS.map(
+								( filter: { slug: string; label: string } ) => (
+									<Tabs.Tab
+										tabId={ filter.slug }
+										key={ filter.slug }
+									>
+										{ filter.label }
+									</Tabs.Tab>
+								)
+							) }
+						</Tabs.TabList>
+					</Tabs>
+				</div>
+			) }
+			<DataViews
+				data={ posts }
+				fields={ fields }
+				view={ view }
+				onChangeView={ onChangeView }
+				actions={ actions }
+				isLoading={ isResolving }
+				paginationInfo={ {
+					totalItems,
+					totalPages,
+				} }
+				defaultLayouts={ DEFAULT_LAYOUTS }
+				getItemId={ getItemId }
+				getItemLevel={ getItemLevel }
+				selection={ selection }
+				onChangeSelection={ ( items: string[] ) => {
+					navigate( {
+						search: {
+							...searchParams,
+							postIds: items.length > 0 ? items : undefined,
+							edit:
+								items.length === 0
+									? undefined
+									: searchParams.edit,
+						},
+					} );
+				} }
+				renderItemLink={ ( { item, ...props }: { item: Post } ) => (
+					<Link
+						to={ `/types/${ postType }/edit/${ encodeURIComponent(
+							item.id
+						) }` }
+						{ ...props }
+						onClick={ ( event ) => {
+							// Temporary fix to prevent triggering
+							// onChangeSelection, which would override the URL.
+							event.stopPropagation();
+						} }
```

Comment:
Copied this solution as well, but I'm wondering, shouldn't this be covered by https://github.com/WordPress/gutenberg/pull/72081 already?